### ISSUE TITLE
Tests fail with pandoc 1.12.3

### ISSUE THREAD

**Issue by opoplawski** [bebraw/pypandoc#124]
Run pypandoc 1.3.3 tests on EL7 with pandoc 1.12.3 fails with:
```
ERROR: test_convert_with_custom_writer (__main__.TestPypandoc)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "tests.py", line 170, in test_convert_with_custom_writer
    received = pypandoc.convert_file(file_name, lua_file_name)
  File "/builddir/build/BUILD/pypandoc-1.3.3/pypandoc/__init__.py", line 140, in convert_file
    outputfile=outputfile, filters=filters)
  File "/builddir/build/BUILD/pypandoc-1.3.3/pypandoc/__init__.py", line 325, in _convert_input
    'Pandoc died with exitcode "%s" during conversion: %s, running %s' % (p.returncode, stderr, args)
RuntimeError: Pandoc died with exitcode "1" during conversion: pandoc: /tmp/tmpidtlmx.lua: openFile: does not exist (No such file or directory)
, running ['/usr/bin/pandoc', '--from=markdown', '--to=/tmp/tmpiDTlmx.lua', '/tmp/tmpGvyRos.md']
```
It looks like pandoc is converting the to filename to lowercase.  I'm not really sure there is much to be done here or if this old version can be supported.  It does not appear to be possible to influence the names used by NamedTemporaryFile.

**Comment by janschulz** [comment]
This seems to be the costume writer functionality which was added in  https://github.com/bebraw/pypandoc/pull/109

My guess is that a newer pandoc version is required to use that. Maybe @snay can help here?

**Comment by snay** [comment]
Custom writers were introduced in 1.12. I've tested different configurations and it seems strange to me. CentOS 7 + Python 2.7.x + Pandoc 1.12.3/1.12.3.1 - i can reproduce the error, with the same configuration and Py3 - can not, all works fine.
Pandoc 1.12.3 + MacOS + both major versions of python - all is fine.
For me, it looks like something OS-dependent changed inside Pandoc between 1.12.x and 1.13. 
If you've suggestions how to localize the bug, please tell me, because i'm out of ideas now.
P.S. It looks like i still had to change test a bit for another matter. WIll check

**Comment by bebraw** [comment]
Could this be an upstream bug? Is it worth reporting to pandoc to see what they say?

**Comment by snay** [comment]
@bebraw Looks like i've found it. 
https://github.com/jgm/pandoc/issues/1267

**Comment by opoplawski** [comment]
Unfortunately it looks like it will take some work to update pandoc to 1.12.4 in EPEL7 - see https://bugzilla.redhat.com/show_bug.cgi?id=1399334

**Comment by bebraw** [comment]
What do you want to do with this issue? Maybe one option would be to do a version specific monkey patch. Not pretty but if it works, I think we could accept a PR.

**Comment by janschulz** [comment]
Isn't it the easiest just to skip the Test (and maybe warn / Error when using the functionality) when pandoc is too old?

**Comment by opoplawski** [comment]
Yeah, that's what I've done now in the Fedora package.  I'm leaving this open in case you want to handle this automatically upstream somehow.  Otherwise, feel free to close.  Thank you for all of the help.

**Comment by janschulz** [comment]
@opoplawski: Could you send the patch as a PR?

**Comment by keszybz** [comment]
FTR, with pandoc-1.19.1-3.fc27.x86_64, the only thing that fails is `test_basic_conversion_from_http_url`.

**Comment by jankatins** [comment]
This should be fixed/worked around by #156
